File: generated-code/ruby/codecraft/TcpReadWrite/model/repair_action.rb
module Model

# Repair action
class RepairAction
    # Target entity's ID
    attr_accessor :target

    def initialize(target)
        @target = target
    end

    # Read RepairAction from input stream
    def self.read_from(stream)
        target = stream.read_int()
        RepairAction.new(target)
    end

    # Write RepairAction to output stream
    def write_to(stream)
        stream.write_int(@target)
    end

    def to_s
        string_result = "RepairAction { "
        string_result += "target: "
        string_result += @target.to_s
        string_result += " }"
        string_result
    end

    def to_str
        to_s
    end
end

end

File: generated-code/python/codecraft/TcpReadWrite/model/repair_action.py
from stream_wrapper import StreamWrapper

class RepairAction:
    """Repair action"""

    __slots__ = ("target",)

    target: int

    def __init__(self, target: int):
        self.target = target
        """Target entity's ID"""

    @staticmethod
    def read_from(stream: StreamWrapper) -> "RepairAction":
        """Read RepairAction from input stream
        """
        target = stream.read_int()
        return RepairAction(target)
    
    def write_to(self, stream: StreamWrapper):
        """Write RepairAction to output stream
        """
        stream.write_int(self.target)
    
    def __repr__(self):
        return "RepairAction(" + \
            repr(self.target) + \
            ")"